### DISCUSSION TITLE
Differences in the n_lags parameter

### DISCUSSION THREAD

**Discussion by simonsapiel** [ourownstory/neural_prophet#1423] [Q&A - get help using NeuralProphet] [answered]
Hi,

I'm not sure I understand the lagged regressors in the model. What is the difference between running something like

```python
m = NeuralProphet(
    quantiles=quantiles,
    yearly_seasonality=True,
    weekly_seasonality="auto",
    daily_seasonality="auto",
    learning_rate=0.02 
)

m.set_plotting_backend("plotly")
m.add_lagged_regressor("traffic_amount", n_lags=24)
metrics = m.fit(df)
```
and

```python
m = NeuralProphet(
    quantiles=quantiles,
    yearly_seasonality=True,
    weekly_seasonality="auto",
    daily_seasonality="auto",
    learning_rate=0.02,
    n_lags=24
)

m.set_plotting_backend("plotly")
m.add_lagged_regressor("traffic_amount")
metrics = m.fit(df)
``` 
specifically what does the n_lags parameter mean when used with the add_lagged_regressor function?

**Comment by ourownstory** [comment, marked as answer]
Hi @simonsapiel, 
The `n_lags=24` set in your second example refers to the series itself - it sets the number of auto-regressive terms.

The `m.add_lagged_regressor("traffic_amount", n_lags=24)` line adds an exogenous variable to be regressed over. Here, `n_lags=24` sets the number of lagged observations of this variable to be included in the regression. (If `n_lags` is not set here, it defaults to same `n_lags` as set for AR).

**Reply by simonsapiel** [reply]
Thanks for the quick reply! I'm having some difficulties with this. For some reason after using this `m.add_lagged_regressor("traffic_amount", n_lags=24)` the `future = m.make_future_dataframe(df, n_historic_predictions=True, periods=365)` only creates a single new row to the existing dataframe to forecast while previously this succesfully created 365 days of new rows. What is it that I'm not understanding here?

**Reply by ourownstory** [reply]
Hi @simonsapiel 
When using AR or lagged regressors, the model only predicts `n_forecasts` steps, exactly, as each forecast step is an individual prediction target. If you do not specify `n_forecasts`, it is defaulted to `n_forecasts=1`.

Tip:  Set `n_forecasts` to a value of about half, or at most equal to `n_lags`. In your case, `24`. If you want to have multiple predictions to validate over, use a holdout set and predict over it (instead of using `make_future_dataframe`  - which is meant for predicting the next unknown period only).

**Reply by simonsapiel** [reply]
@ourownstory Oh that’s a bummer. I was under the impression that I could use the lagged feature and still forecast for 365 days since that’s what I need. I have additional future regressor for which I have the needed 365 future values, but I thought that adding this lagged regressor would increase the accuracy for the first 24 days ahead and then the forecast would rely on the future regressor for the remaining horizon. I need to review my knowledge about lags if this is indeed the case.
